avm/res/azure-stack-hci/cluster/tests/e2e/2nodeswitched.defaults/main.test.bicep

```diff
@@ -0,0 +1,304 @@
+      jumboPacket: '9014'
+      networkDirect: 'Disabled'
+      networkDirectTechnology: 'iWARP'
+    }
+    overrideQosPolicy: true
+    qosPolicyOverrides: {
+      bandwidthPercentage_SMB: '50'
+      priorityValue8021Action_Cluster: '7'
+      priorityValue8021Action_SMB: '3'
+    }
+    overrideVirtualSwitchConfiguration: false
+    virtualSwitchConfigurationOverrides: {
+      enableIov: 'true'
+      loadBalancingAlgorithm: 'Dynamic'
+    }
+    trafficType: ['Storage']
+  }
+]
+@description('Optional. Enable storage auto IP configuration. If false, storageNetworks must include IP configurations.')
+param enableStorageAutoIp bool = true
+@description('Optional. The storage networks for the cluster.')
+param storageNetworks storageNetworksArrayType = [
+  {
+    adapterName: 'smb0'
+    vlan: '711'
+  }
+  {
+    adapterName: 'smb1'
+    vlan: '712'
+  }
+]
+param deployArcGateway bool = false
+
+#disable-next-line no-hardcoded-location // Due to quotas and capacity challenges, this region must be used in the AVM testing subscription
+var enforcedLocation = 'southeastasia'
+
+var clusterWitnessStorageAccountName = '${take('${deploymentPrefix}${serviceShort}${take(uniqueString(resourceGroup.id,resourceGroup.location),6)}',21)}wit'
+var keyVaultDiagnosticStorageAccountName = '${take('${deploymentPrefix}${serviceShort}${take(uniqueString(resourceGroup.id,resourceGroup.location),6)}',21)}kvd'
+var keyVaultName = 'kvhci-${deploymentPrefix}${take(uniqueString(resourceGroup.id,resourceGroup.location),6)}'
+
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2024-03-01' = {
+  name: resourceGroupName
+  location: enforcedLocation
+}
+
+module hciDependencies 'dependencies.bicep' = {
+  name: '${uniqueString(deployment().name, enforcedLocation)}-test-hcidependencies-${serviceShort}'
+  scope: resourceGroup
+  params: {
+    clusterNodeNames: clusterNodeNames
+    clusterWitnessStorageAccountName: clusterWitnessStorageAccountName
+    deployArcGateway: deployArcGateway
+    deploymentPrefix: deploymentPrefix
+    deploymentUsername: deploymentUsername
+    deploymentUserPassword: localAdminAndDeploymentUserPass
+    domainOUPath: domainOUPath
+    hciResourceProviderObjectId: hciResourceProviderObjectId
+    hciHostAssignPublicIp: hciHostAssignPublicIp
+    keyVaultName: keyVaultName
+    keyVaultDiagnosticStorageAccountName: keyVaultDiagnosticStorageAccountName
+    localAdminPassword: localAdminAndDeploymentUserPass
+    localAdminUsername: localAdminUsername
+    location: enforcedLocation
+    arbDeploymentAppId: arbDeploymentAppId
+    arbDeploymentSPObjectId: arbDeploymentSPObjectId
+    arbDeploymentServicePrincipalSecret: arbDeploymentServicePrincipalSecret
+    vnetSubnetId: vnetSubnetId
+    hciNodeCount: length(clusterNodeNames)
+    switchlessStorageConfig: false
+    hciISODownloadURL: hciISODownloadURL
+    hciVHDXDownloadURL: hciVHDXDownloadURL
+  }
+}
+
+module cluster_validate '../../../main.bicep' = {
```

Comment:
There should only be one test deployment in this file. If you need dependencies for yhe cluster deployment, they should be deployed in the dependencies.bicep file(s) and **not.** use the module's code.

Reply:
@AlexanderSehr  This is due to an unconventional use of resources, where to have a fully deployed cluster, you first submit deploymentSettings with deploymentMode = 'Validate'. Once that is successfully validated, you resubmit the same deployment setting with deploymentMode = 'Deploy'. using the module, I could call the same resource twice. 

Is there a better way you recommend?